File: fm-dddrive/src/main/java/io/dddrive/core/ddd/model/enums/CodePartListTypeEnum.java
package io.dddrive.core.ddd.model.enums;

import io.dddrive.core.enums.model.base.EnumerationBase;

public class CodePartListTypeEnum extends EnumerationBase<CodePartListType> {

	private static CodePartListTypeEnum INSTANCE;

	public CodePartListTypeEnum() {
		super(CodePartListType.class);
		INSTANCE = this;
	}

	public static CodePartListTypeEnum getInstance() {
		return INSTANCE;
	}

	@Override
	public void addItem(CodePartListType item) {
		super.addItem(item);
	}

	public static CodePartListType getPartListType(String partListTypeId) {
		return INSTANCE.getItem(partListTypeId);
	}

}

File: fm-dddrive/src/main/java/io/dddrive/core/ddd/model/enums/CodePartListTypeEnum.kt
package io.dddrive.core.ddd.model.enums

import io.dddrive.core.enums.model.base.EnumerationBase

class CodePartListTypeEnum : EnumerationBase<CodePartListType>(CodePartListType::class.java) {

	init {
		instance = this
	}

	companion object {

		lateinit var instance: CodePartListTypeEnum

		@JvmStatic
		fun getPartListType(partListTypeId: String): CodePartListType = instance.getItem(partListTypeId)

	}

}
